packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -424,6 +429,8 @@ class StorageS3Service {
     } on smithy.UnknownSmithyHttpException catch (error) {
       // S3Client.copyObject may return 403 or 404 error
       throw S3Exception.fromUnknownSmithyHttpException(error);
+    } on AWSHttpException catch (error) {
```

Comment:
Instead of duplicating the same pattern 'on smithy throw, on awsHttpexception, throw' can you wrap this in common code.  For Analytics I did that in a `pinpointAnalyticsException.fromPinpointException` method.

Reply:
I personally prefer to keep the exception catching inline and close to the API call so I'd know immediately what kind of exception I would expect from the API call.